src/client/Microsoft.Identity.Client/AppConfig/CacheOptions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Options for MSAL token caches. 
+    /// </summary>
+    /// <remarks>
+    /// These option do not allow configuration of external cache serialization, for which you should use <see cref="TokenCache.SetAfterAccessAsync(System.Func{TokenCacheNotificationArgs, System.Threading.Tasks.Task})"/> and other callbacks.
+    /// For detailed recommendations see: https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-token-cache-serialization?tabs=aspnetcore
+    /// </remarks>
+    public class CacheOptions
```

Comment:
Suggestion to simplify the API. Use factory pattern. So users know what they create i.e. a static method to create options and all constructors as private. 
`public class CacheOptions
{
     public static CacheOptions CreateCacheOption(bool useSharedCache) {...}
     private CacheOptions(bool useSharedCache) {...}
     bool UseSharedCache { get; }
  
}`

Reply:
Another way can be declare CacheOptions as enum. So it can be expanded after preview. and it will make API simpler.

Reply:
We do have other options objects (e.g. SystemWebviewOptions) and they are all simple CRUD objects. So for consistency I'd keep it the same. Also, there is currently 1 option and we don't know how many more to come, but I estimate no more than 3.